File: backend/app/schemas/offseason.py
from pydantic import BaseModel, ConfigDict
from typing import List, Optional

class TeamNeed(BaseModel):
    position: str
    current_count: int
    target_count: int
    need_score: float  # Higher means more needed

class Prospect(BaseModel):
    id: int
    name: str
    position: str
    overall_rating: int
    
    model_config = ConfigDict(from_attributes=True)

class DraftPickSummary(BaseModel):
    round: int
    pick_number: int
    team_id: int
    player_name: str
    player_position: str
    player_overall: int

class DraftPickDetail(BaseModel):
    id: int
    season_id: int
    team_id: int
    original_team_id: int
    round: int
    pick_number: int
    player_id: Optional[int] = None
    
    model_config = ConfigDict(from_attributes=True)

class PlayerProgressionResult(BaseModel):
    player_id: int
    name: str
    position: str
    change: int
    old_rating: int
    new_rating: int
    
    model_config = ConfigDict(from_attributes=True)

class PlayerProgressionSummary(BaseModel):
    player_id: int
    player_name: str
    previous_rating: int
    new_rating: int
    change: int

class FreeAgentSigning(BaseModel):
    player_id: int
    player_name: str
    position: str
    overall_rating: int
    age: int
    team_id: int
    team_name: str
    years: int = 1
    total_salary: float = 0.0
    aav: float = 0.0
    guaranteed_money: float = 0.0
    grade: str = "B"
    competing_offers_count: int = 1
    
    # Optional alias fields
    contract_years: Optional[int] = None
    total_value: Optional[float] = None
    annual_avg: Optional[float] = None
    guaranteed: Optional[float] = None
    signing_grade: Optional[str] = None
    signing_round: Optional[int] = None
    bidding_teams_count: Optional[int] = None
    
    model_config = ConfigDict(from_attributes=True, populate_by_name=True)

    def __init__(self, **data):
        if "contract_years" in data and "years" not in data:
            data["years"] = data["contract_years"]
        if "total_value" in data and "total_salary" not in data:
            data["total_salary"] = data["total_value"]
        if "annual_avg" in data and "aav" not in data:
            data["aav"] = data["annual_avg"]
        if "guaranteed" in data and "guaranteed_money" not in data:
            data["guaranteed_money"] = data["guaranteed"]
        if "signing_grade" in data and "grade" not in data:
            data["grade"] = data["signing_grade"]
        if "bidding_teams_count" in data and "competing_offers_count" not in data:
            data["competing_offers_count"] = data["bidding_teams_count"]
        super().__init__(**data)

class FreeAgentMarketPlayer(BaseModel):
    player_id: int
    player_name: str
    position: str
    overall_rating: int
    age: int
    projected_aav: float
    projected_years: int
    tier: str
    top_interested_teams: List[str]
    
    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/offseason.ts
import type { CombineResult } from "./combine";

export interface TeamNeed {
  position: string;
  current_count: number;
  target_count: number;
  need_score: number;
  priority?: "high" | "medium" | "low";
  starter_quality?: number;
  league_avg_quality?: number;
  depth_breakdown?: {
    starters: number;
    backups: number;
  };
}

export interface Prospect {
  id: number;
  name: string; // This might need to be split or handled if backend sends first/last
  first_name: string;
  last_name: string;
  position: string;
  college?: string;
  overall_rating: number;
  height: number;
  weight: number;
  age: number;
  speed: number;
  acceleration: number;
  strength: number;
  agility: number;
  projected_round?: number;
  combine?: CombineResult;
  genesis_revealed?: boolean;
  scouting_report?: ProspectScoutingReport;
  visual_assets?: {
    headshot: string;
    hero_pose: string;
    action_pose: string;
    celebration: string;
  };
}

export interface ProspectScoutingReport {
  prospect_id: string;
  completion: number;
  attributes: Record<
    string,
    {
      value: number | null;
      range: [number, number] | null;
      tier: string;
      display: string;
    }
  >;
  strengths: string[];
  weaknesses: string[];
}

export type ScoutingReport = ProspectScoutingReport;

export interface DraftPickSummary {
  round: number;
  pick_number: number;
  team_id: number;
  player_name: string;
  player_position: string;
  player_overall: number;
}

export interface DraftPickDetail {
  id: number;
  season_id: number;
  team_id: number;
  original_team_id: number;
  round: number;
  pick_number: number;
  player_id?: number;
}

export interface PlayerProgressionResult {
  player_id: number;
  name: string;
  position: string;
  change: number;
  old_rating: number;
  new_rating: number;
}

export interface SalaryCapData {
  team_id: number;
  team_name: string;
  total_cap: number;
  used_cap: number;
  available_cap: number;
  cap_percentage: number;
  top_contracts: {
    player_id: number;
    name: string;
    position: string;
    salary: number;
    years_left: number;
  }[];
  position_breakdown: {
    group: string;
    total_salary: number;
    percentage: number;
  }[];
  league_avg_available: number;
  projected_rookie_impact: number;
}

/**
 * Genesis Reveal data returned from the combine/genesis-reveal endpoint (B-047).
 * Contains revealed "true" stats and traits for a prospect.
 */
export interface GenesisRevealData {
  player_id: number;
  position: string;
  revealed_stats: Record<string, number>;
  revealed_traits: string[];
  confidence_level: number;
  scouting_accuracy: number;
}

export interface FreeAgentSigning {
  player_id: number;
  player_name: string;
  position: string;
  overall_rating: number;
  age: number;
  team_id: number;
  team_name: string;
  years: number;
  total_salary: number;
  aav: number;
  guaranteed_money: number;
  grade: string;
  competing_offers_count: number;
  contract_years?: number;
  total_value?: number;
  annual_avg?: number;
  guaranteed?: number;
  signing_grade?: string;
  signing_round?: number;
  bidding_teams_count?: number;
}

export interface FreeAgentMarketPlayer {
  player_id: number;
  player_name: string;
  position: string;
  overall_rating: number;
  age: number;
  projected_aav: number;
  projected_years: number;
  tier: string;
  top_interested_teams: string[];
  name?: string;
  experience?: number;
  projected_market_value?: number;
}

export interface FreeAgentBidRequest {
  player_id: number;
  team_id: number;
  years: number;
  total_amount: number;
  signing_bonus: number;
  guaranteed_amount: number;
}

export interface FreeAgentBidResponse {
  status: string;
  accepted: boolean;
  message: string;
  updated_cap_space: number;
}
